docs/core/tutorials/netcore-hosting.md

```diff
@@ -19,40 +19,127 @@ Because hosts are native applications, this tutorial will cover constructing a C
+*  `APP_NI_PATHS` This list is very similar to APP_PATHS except that it's meant to be paths that will be probed for native images.
+*  `NATIVE_DLL_SEARCH_DIRECTORIES` This property is a list of paths the loader should probe when looking for native DLLs called via p/invoke.
+*  `PLATFORM_RESOURCE_ROOTS` This list includes paths to probe in for resource satellite assemblies (in culture-specific sub-directories).
+
+In our sample host, the TPA list is constructed by simply listing all libraries in the current directory:
+
+[!code-cpp[CoreClrHost#7](../../../samples/core/hosting/HostWithCoreClrHost/SampleHost.cpp#7)]
+
+Because the sample is simple, we only need the `TRUSTED_PLATFORM_ASSEMBLIES` and `APP_PATHS` properties:
+
+[!code-cpp[CoreClrHost#3](../../../samples/core/hosting/HostWithCoreClrHost/SampleHost.cpp#3)]
+
+### Step 4 - Start the runtime and default AppDomain
+
+Now we're ready to initialize the .NET Core runtime. Unlike the mscoree.h hosting API (described below), CoreCLRHost.h APIs start the runtime and create the default AppDomain all with a single call. We use `coreclr_initialize`, specifying a base path, name, and properties for the default AppDomain and receiving back the host handle and AppDomain ID.
+
+[!code-cpp[CoreClrHost#4](../../../samples/core/hosting/HostWithCoreClrHost/SampleHost.cpp#4)]
+
+### Step 5 - Run managed code!
+
+With the runtime started and an AppDomain initialized, we're ready to call managed code. This can be done a couple different ways. The sample code linked to this tutorial uses the `coreclr_create_delegate` function to create a delegate to a static managed method. This API takes the assembly name, namespace-qualified type name, and method name as inputs and returns a delegate that can be used to invoke the method.
+
+[!code-cpp[CoreClrHost#5](../../../samples/core/hosting/HostWithCoreClrHost/SampleHost.cpp#5)]
+
+In this sample, the host can now call `managedDelegate` to run the `ManagedWorker.DoWork` method.
+
+Alternatively, the `coreclr_execute_assembly` function can be used to launch a managed executable. This API takes an assembly path and arrray of arguments as input parameters. It loads the assembly at that path and invokes its main method. 
+
+```C++
+int hr = executeAssembly(
+        hostHandle,
+        domainId,
+        argumentCount,
+        arguments,
+        "HelloWorld.exe",
+        (unsigned int*)&exitCode);
+```
+
+### Step 6 - Shutdown and clean up
+
+Finally, when the host is done running managed code, the .NET Core runtime is shut down (and AppDomains unloaded) with the `coreclr_shutdown` function (or `coreclr_shutdown_2` which is the same, but retrieves the managed code's exit code).
```

Comment:
Remove documentation regarding `coreclr_shutdown_2` - that statement is made above.

Reply:
Good point. I removed the explanation of what coreclr_shutdown_2 does and just listed both functions as options here.